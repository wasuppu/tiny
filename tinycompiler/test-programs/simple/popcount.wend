main() {
    int test;

    // count number of set bits in binary representation of a positive integer n
    int popcount(int n) {
        int cnt;

        cnt = 0;
        while n > 0 {
            cnt = cnt + n % 2;
            n = n / 2;
        }
        return cnt;
    }

    test = 534219;
    print "number of 1s in the binary representation of ";
    print test;
    print " is ";
    println popcount(test);
}

