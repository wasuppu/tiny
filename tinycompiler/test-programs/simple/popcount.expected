number of 1s in the binary representation of 534219 is 9
